### ISSUE TITLE
not working on pluralsight.com

### ISSUE THREAD

**Issue by salsamaani** [igrigorik/videospeed#521]
thanks alot, it is a usefull great extention
but since a week i cant apply it on pluralsight.com (i am using chrome)

**Comment by jacobcolbert** [comment]
Can you find a video that doesn't require creating an account i could test on?

**Comment by thewheat** [comment]
This video can be accessed without a login and shows the problem
https://app.pluralsight.com/course-player?clipId=1e8b695e-a148-4b6b-83d0-07c9ddbb9b73

![image](https://user-images.githubusercontent.com/892961/66249565-e28c3d00-e767-11e9-9bac-9bfb721e1246.png)

Could probably be something to do with their own speed controls
![image](https://user-images.githubusercontent.com/892961/66249613-c9d05700-e768-11e9-898c-e985bd985975.png)

Strange enough this link allows the video speed to be maintained https://app.pluralsight.com/player?name=88240101-1955-4c4c-8404-653a2fc3f135&mode=live&clip=0&course=aws-security-introduction&author=jordan-yankovich 
- from the URL path it uses `player` instead of `course-player`
- logged in users always get redirected to `course-player` so they can't manually go to `player`

**Demo of the problem**
![pluralsight](https://user-images.githubusercontent.com/892961/66249545-ace75400-e767-11e9-809c-238cbc299bda.gif)

**Comment by thewheat** [comment]
Dug in a bit more and there is a `ratechange` event listener

![image](https://user-images.githubusercontent.com/892961/66696850-264feb00-ed02-11e9-9494-8ee8dd2234e6.png)

If you remove it, it will work (hover over it in dev tools and a "Remove" button will show to remove it)
![image](https://user-images.githubusercontent.com/892961/66696934-2997a680-ed03-11e9-8fd5-075e007ce72f.png)


Looking at some of the code it looks like there is some stored state which VSC doesn't modify and thus it resets back to the stored playbackRate value 
```
  onRateChange = rate => {
    const { current: player } = this.element

    player && this.setState({ reportedPlaybackRate: player.playbackRate })
  }
```
...
```
    if (nextProps.playbackRate !== reportedPlaybackRate) {
      this.setPlaybackRate(nextProps.playbackRate)
    }
```

**Comment by ChadBailey** [comment]
Currently working on this one... I'll use this thread since it's the original and has more details. Here's my post from #531 

> Unfortunately, this bug highlights a striking limitation of JavaScript. There is no way to remove an event listener if you can't supply the original function to your removal command. In the case of PluralSight that's an anonymous function which makes it literally impossible to remove.
> 
> I'm not saying this bug can't be fixed, but I am saying it can't be fixed without taking some pretty ugly hacky approach. I worked at it for a good while and was unable to find any effective solution for stripping or overloading the event listener that sets the playback rate back as soon as it's altered.

My progress so far...

- Confirmed can visit video element in inspector and remove "ratechange" event listener to manually resolve issue
- Searched for an inordinate amount of time looking for a way to remove an event listener programmatically (without access to the original function), every place I found was a dead end
- Only hopeful way I found to resolve was to block event listeners using the technique found here: https://codeday.me/en/qa/20190305/3259.html
- Confirmed I can make keybinds work (at the expense of the clickable controller not working) via pasting the following snippet into the console - inspired by aforementioned link

```javascript
    document.body.addEventListener("ratechange", function (event) {
      event.stopPropagation();
    }, true);
```

- Tried adding this snippet to various parts of the code to mixed results (it seemed to work once, but could not reproduce). Generally it just completely breaks playback rate adjustment
- Tried updating existing ratechange event listener to include `event.stopPropagation();` and adding the `true` switch. This seems to have no effect at all
- Tried modifying the above statement to add only to video element instead of the entire body which seemed to give roughly the same results

Not sure what else to try... keep in mind I have limited JS experience, so it's possible I'm missing something obvious
